sherpa/utils/__init__.py

```diff
@@ -1452,51 +1457,144 @@ def create_expr(vals, mask=None, format='%s', delim='-'):
         index = numpy.arange(len(mask))
         seq = index[mask]
 
-    # diffs has 1 less element than vals
-    #
-    diffs = numpy.apply_along_axis(numpy.diff, 0, seq)
-    diffs = diffs == 1
-
-    def filt(start, end):
-        "What is the filter expression for this range?"
-        vstr = format % start
-        if start != end:
-            vstr += delim + format % end
-        return vstr
-
-    # The trick here is that if diffs is True then we just want to
-    # update the end counter, which gives the end-point of the current
-    # range, and move to the next bin. We only output data when diffs
-    # is False, when we report the previous range and start a new
-    # range.
-    #
-    expr = []
+    exprs = []
     start = vals[0]
-    end = vals[0]
-    for delta, val in zip(diffs, vals[1:]):
 
-        # If this is part of a contiguous sequence then we update
-        # the end value and move on.
-        #
-        if delta:
-            end = val
-            continue
+    # We follow create_expr_int but instead of having separate lo/hi
+    # always we use the same array
+    #
+    startbins = vals[1:]
+    endbins = vals[:-1]
 
-        # If we have a break then output the range we have just
-        # processed.
-        #
-        expr.append(filt(start, end))
+    diffs = numpy.diff(seq)
+    idxs, = numpy.where(diffs != 1)
+    for idx in idxs:
+        exprs.append((start, endbins[idx]))
+        start = startbins[idx]
 
-        # Start a new range.
-        #
-        start = val
-        end = val
+    exprs.append((start, vals[-1]))
+
+    def filt(lo, hi):
+        vstr = format % lo
+        if lo == hi:
+            return vstr
+
+        return vstr + f"{delim}{format % hi}"
+
+    return ",".join([filt(*expr) for expr in exprs])
+
+
+def create_expr_int(lovals, hivals, mask=None,
```

Comment:
The same applies to `create_expr`, but that is existing code that uses the "mask" terminology. I think it should be renamed there as well to fix this, while keeping both functions consistent.

Reply:
I can rename `create_expr_int` as it is new in this PR. I'm more reluctant to change `create_expr` in part because I don't know what you suggest the new name should be.

Reply:
I suggest to leave `create_expr` as is and rename `create_expr_int` to `create_expr_integrated`